### PR TITLE
Fixes standard RPEDs not working on machines

### LINKED ISSUE #82525: All RPED interactions broken.
Reporting client version: 515.1630

## Issue Summary

Replacing parts and checking parts of any device do not work. Parts can still be inserted into the RPED, but without any effect.

## Round ID:
[226779](https://scrubby.melonmesa.com/round/226779)

### PR DESCRIPTION
**Pull request by 00-Steven**
## About The Pull Request

Previously, `exchange_parts(...)` would cancel if both the `NO_DECONSTRUCTION` flag was set and you couldn't use your part replacer from a distance.
https://github.com/tgstation/tgstation/blob/1583cf0cc968cd9f5da2398035feb8d70d58bcf2/code/game/machinery/_machinery.dm#L958-L959
Our recent removal of `NO_DECONSTRUCTION`, however, has left this to _only_ be the latter.
https://github.com/tgstation/tgstation/blob/f0ed4ba4ce6b114509c10ee2f36ab0af6d7c81d2/code/game/machinery/_machinery.dm#L956-L957
Buuuuut this makes it unconditionally cancel for normal RPEDs, instead of only blocking them if `NO_DECONSTRUCTION` was set.

As `NO_DECONSTRUCTION` is very much no longer relevant for this purpose, we simply remove the ranged RPED check altogether.
This fixes our issue.
## Why It's Good For The Game

Fixes #82525.
## Changelog
:cl:
fix: Standard RPEDs work on machines again.
/:cl:

### CODE REVIEW

**Comment by 00-Steven** [comment]
image
...never had that vne happen before, what the hell?

File: code/game/machinery/_machinery.dm [hunk 1]
```diff
@@ -953,9 +953,6 @@
 	if(!istype(replacer_tool))
 		return FALSE
```
Comment (ZephyrTFA) [inline code/game/machinery/_machinery.dm:955]:
```suggestion
	if(!user.Adjacent(src) && !replacer_tool.works_from_distance)
		return FALSE
```

Comment (ZephyrTFA) [reply code/game/machinery/_machinery.dm:955]: keep the ranged check; but make it only matter if we are ranged

Comment (00-Steven) [reply code/game/machinery/_machinery.dm:955]: Are you sure? I believe this is handled on the RPED side already
https://github.com/tgstation/tgstation/blob/9d135c67fd4a1e6bbfb22cb0da8a3bf49516f16d/code/modules/research/stock_parts.dm#L42-L43

Comment (00-Steven) [reply code/game/machinery/_machinery.dm:955]: This _has_ brought to my attention that vending machines currently seem to have a similar issue going on, I'll look into it tomorrow

Comment (00-Steven) [reply code/game/machinery/_machinery.dm:955]: yeaaaaaah seems like they got the same treatment
now:
https://github.com/tgstation/tgstation/blob/9d135c67fd4a1e6bbfb22cb0da8a3bf49516f16d/code/modules/vending/_vending.dm#L1145-L1146
before:
https://github.com/tgstation/tgstation/blob/c1f11f26cef91e7279bd4ac936274f78151ad9b0/code/modules/vending/_vending.dm#L1145-L1146

Comment (SyncIt21) [reply code/game/machinery/_machinery.dm:955]: Fix vending machines too since it's a simple one liner

**Comment by Byzz2077** [comment]
someone merge this already good lord

File: code/game/machinery/_machinery.dm (hunk 1 above)
Comment (00-Steven) [reply code/game/machinery/_machinery.dm:955]: yeah gimme a bit, had some Poorly Timed Responsibilities

**Comment by 00-Steven** [comment]
amazing timing lmao

**Comment by Byzz2077** [comment]
big thanks for fixing this! <3

**Pull request merged**